spec/src/main/asciidoc/overview.asciidoc

```diff
@@ -79,12 +79,52 @@ Instances of `ManagedExecutor` and `ThreadContext` can be injected into CDI bean
     unmanagedCompletionStage.thenApply(threadContext.contextualFunction(function3));
 ----
 
-The container provides default instances of `ManagedExecutor` and `ThreadContext`, as injected above, which are shared within an application with all other injection points for `ManagedExecutor` and `ThreadContext` that do not specify a qualifier. Their configuration is equivalent to `ManagedExecutor.builder().build()` and `ThreadContext.builder().build()`.
+The container allows to inject default instances of `ManagedExecutor` and `ThreadContext` which do not leverave either `@NamedInstance` or `ManagedExecutorConfig`/`ThreadContextConfig` annotations, as shown above.
```

Comment:
In the absence of other qualifiers and annotations, the container creates and injects a new default instance of `ManagedExecutor` or `ThreadContext` per injection point, as shown above.

Reply:
I am in favor of having `@Default` injection points use a shared instance, rather than one instance per injection point.  Conceptually I think of `@NamedInstance` as qualifier, and `@XXConfig` being a "synthetic producer" for the associated qualifier (whether the qualifier is explicitly declared or inferred via FQCN+field).